### ISSUE TITLE
Finish up rv6 filesystem

### ISSUE THREAD

**Issue by tjhu** [mars-research/redleaf#9]
* [x] Support mapping devices as files.
* [x] Support pipes.
* [x] `sleep()` in `begin_op()`. Blocked by #7.
* [x] 3 layer inode indirection so we can have 1GB large files(currently 2 layers).
* [x] Rewrite `BufferCache` so that's it's safer, faster, and easier to use.
* [ ] Speed up I/O by utilizing the AIO and large block size interfaces exposed by the `BDev`. (quite complicated; maybe create a new issue to track this; see how biscuit does it).
* [x] Investigate the usage of `begin/end_op()`. Maybe we could refactor it in a cleaner and safer way using RAII or session-based methods.
